pkg/api/validation/validation.go

```diff
@@ -1954,12 +1954,8 @@ func ValidateVolumeMounts(mounts []api.VolumeMount, volumes sets.String, contain
 		if mountpoints.Has(mnt.MountPath) {
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("mountPath"), mnt.MountPath, "must be unique"))
 		}
-		if !path.IsAbs(mnt.MountPath) {
-			// also allow windows absolute path
-			p := mnt.MountPath
-			if len(p) < 2 || ((p[0] < 'A' || p[0] > 'Z') && (p[0] < 'a' || p[0] > 'z')) || p[1] != ':' {
-				allErrs = append(allErrs, field.Invalid(idxPath.Child("mountPath"), mnt.MountPath, "must be an absolute path"))
-			}
+		if !filepath.IsAbs(mnt.MountPath) {
```

Comment:
This presumes that the server on which the validation runs is the same OS as the server on which the pod will run.  That is certainly not guaranteed.

My gut feeling is we should just remove this validation, and define that any path which is not absolute (however the target node defines that) is rooted at / (linux) or c:\ (windows).  I need help thinking through potential implications of that, but I can't think of any off the top of my head.

Obviously this is a bigger change, because it means kubelet has to adapt.  And it needs tests, probably node e2e.